test/Chainweb/Test/Tools/TXGen.hs

```diff
@@ -0,0 +1,154 @@
+{-# language OverloadedStrings #-}
+
+-- |
+-- Module: Chainweb.Test.Tools.TxGen
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: See LICENSE file
+-- Maintainer: Emmanuel Denloye-Ito <emmanuel@kadena.io>
+-- Stability: experimental
+--
+-- Unit test for Transaction Generator via the Http Pact interface (/send,
+-- etc.) (inprocess) API in Chainweb
+--
+module Chainweb.Test.Tools.TXGen (tests) where
+
+import Control.Monad (replicateM, void)
+import Control.Monad.IO.Class
+
+import Data.Foldable
+import Data.List.NonEmpty (NonEmpty)
+import Data.Map.Strict (Map)
+import Data.Maybe
+import Data.String.Conv
+import qualified Data.Aeson as A
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.List.NonEmpty as NEL
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+
+import Fake
+
+import GHC.Stack
+
+import Numeric.Natural
+
+
+import Servant.Client
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+-- pact imports
+import Pact.Types.API
+import Pact.Types.Capability
+import Pact.Types.ChainMeta
+import Pact.Types.Command
+import Pact.Types.Exp
+import Pact.Types.Info (mkInfo)
+import Pact.Types.Names
+import Pact.Types.PactValue
+
+-- chainweb imports
+import Chainweb.ChainId
+import Chainweb.Graph
+import Chainweb.Test.Pact.RemotePactTest hiding (tests)
+import Chainweb.Test.Pact.Utils
+import Chainweb.Test.Utils
+import Chainweb.Time
+import Chainweb.Utils
+import Chainweb.Version
+
+import Data.CAS.RocksDB
+
+import TXG.Simulate.Contracts.CoinContract
+import TXG.Simulate.Contracts.Common
+
+nNodes :: Natural
+nNodes = 1
+
+cid :: HasCallStack => ChainId
+cid = head . toList $ chainIds v
+
+v :: ChainwebVersion
+v = FastTimedCPM petersonChainGraph
+
+tests :: RocksDb -> ScheduledTest
+tests rdb = testGroupSch "Chainweb.Test.Tools.TxGen"
+    [ withNodes rdb nNodes $ \net ->
+        withMVarResource 0 $ \iomvar ->
+        withTime $ \iot ->
+        testGroup "txgen tests" [
+            withRequestKeys iot iomvar net $ responseGolden net
```

Comment:
What is the purpose of this golden test? Golden test tent to be fragile and are best used to catch and get manual confirmation to changes for things like API data or data that is hashed and put on the chain. Golden tests are less suited for testing functionality.